### ISSUE TITLE
AdGuard tampers with apt requests from "Android OS" and there is no obvious way to allowlist it

### ISSUE THREAD

**Issue by AetherMagee** [AdguardTeam/AdguardForAndroid#4950]
### Please answer the following questions for yourself before submitting an issue.

- [X] I am running the latest version
- [X] I checked the documentation and found no answer
- [X] I checked to make sure that this issue has not already been filed

### AdGuard version

4.2 Nightly 25

### Environment

```markdown
- OS: Android 13
- Device: Xiaomi Redmi Note 8 Pro
```

### Root access

- [X] Yes, I have it.

### Issue Details

Steps to reproduce:
1. Set up a basic Ubuntu-like chroot environment with apt
2. Try to execute any apt command that depends on connecting to repositories, like apt update 

### Expected Behavior

Command running normally

### Actual Behavior

apt is apparently unable to verify the repo's signature and connection attempts return error 400 Bad Request

### Screenshots

<details><summary>Screenshot 1:</summary>

![Screenshot_20230826-183532_AdGuard](https://github.com/AdguardTeam/AdguardForAndroid/assets/59086566/0ce047e1-3ddb-49b7-acb5-879a1d401c93)
Even though the request is "allowlisted", the problem is still present

</details>
<details><summary>Screenshot 2:</summary>

![Screenshot_20230826-184106_Termux~2](https://github.com/AdguardTeam/AdguardForAndroid/assets/59086566/9c2e8294-67ce-43db-afc1-b34dbe7484ec)

</details>

### Additional Information

The problem is present regardless of whether HTTPS filtration is on or off.

Other tools like curl, git, wget are fine, so maybe it is apt's fault, but the issue is only present with AdGuard on.

The problem is not with the custom rule which can be seen on Screenshot 1 (the request is allowlisted), it was added as an attempt to solve the problem and it didn't help.

**Comment by maxikuzmin** [comment]
@AetherMagee I have reproduced this case study, I have a 404 error on the 4th and 5th requests, but this is also repeated without AdGuard.
Without AdGuard, do you have stable apt updates? Have you also tried the command without sudo?

**Comment by AetherMagee** [comment]
@maxikuzmin Yes, with AdGuard turned off apt is working fine. Sudo doesn't change anything in this situation.

**Comment by AetherMagee** [comment]
Here's a screen recording of what's happening on my side

https://github.com/AdguardTeam/AdguardForAndroid/assets/59086566/9ed74fc8-284d-473a-ae15-24dd4b8f402b

**Comment by maxikuzmin** [comment]
@AetherMagee hmm, that's strange because I have the same behavior in both cases.

it would be nice to get your logs.

- Enable debug logging:

Settings -> General -> Advanced -> Logging level -> Debug

- Reproduce the issue related to the button behaviour, then remember the exact time when it happened

- Collect logs:

Settings -> General -> Advanced -> Export logs and system info

- Sent this file to [devteam@adguard.com](mailto:devteam@adguard.com):

mention the 4950 number in the subject
specify the exact time when the issue occurred

**Comment by maxikuzmin** [comment]
@AetherMagee thank you! We've received your logs. We need some time to look into it
